src/control/client/connect.go

```diff
@@ -119,13 +120,17 @@ func (c *connList) ConnectClients(addresses Addresses) ResultMap {
 		c.controllers = append(c.controllers, controller)
 	}
 
+	time.Sleep(1000 * time.Millisecond)
```

Comment:
Is there another way we could synchronize this? Something deterministic to wait for? Sleeps always stick out to me as potential future bugs.

Reply:
yes me too but in this case I can't figure out what we are synchronising with, seems to be something internal with dial . I reduce this time out to 100ms in the next patch. Without this delay, states don't get reflected properly.

Reply:
its not clear to me how this is a race condition. That initial receive on the channel inside the for range addresses should be blocking. You shouldn't be able to get past that point until every address had been received.

Reply:
I spent a while investigating and I don't have an answer, the result of not having a delay is that the underlying state of the connection is not reported as expected

Reply:
I agree that this is definitely not nice ...